### ISSUE TITLE
How to use Moor to set the start value of sqlite sequence?

### ISSUE THREAD

**Issue by jims57** [simolus3/moor#1452]
![image](https://user-images.githubusercontent.com/1920873/132715356-d7279d0c-e43b-4f20-a696-bdabd4aa11a8.png)

By the way, it will be good if Moor can set all these tables' sequence in batch. Thanks.

**Comment by simolus3** [comment]
You could write your own table that mimics `sqlite_sequence`:

```dart
class SqliteSequence extends Table {
  String get tableName => 'sqlite_sequence';
  
  TextColumn get name => text()();
  IntColumn get seq => integer()();
}
```

You'd have to change the migration logic in your database to not create this table of course. But then, you could write updates to that tables like any other table, including in a batch.

I don't see a compelling reason to add this feature into moor given that it will probably only be used rarely and since a safe workaround exists (just don't create the table).

**Comment by jims57** [comment]
> change the migration logic in your database to not create this table of course

By the way, how to make **Moor** not to create **sqlite_sequence** table in migration logic?

**Comment by jims57** [comment]
Moreover, I was trying to use [Custom Query](https://moor.simonbinder.eu/docs/using-sql/custom_queries/#statements-with-a-generated-api) way to set the sequence, why does it fail to build?

![image](https://user-images.githubusercontent.com/1920873/132794608-ba0bfd16-f536-4d44-a745-87a0169a0387.png)


Is it because **Moor** doesn't recognize the **sqlite**'s built-in **sqlite_sequence** table?

My question in this reply is that how to make **Moor** use these built-in tables **Sqlite** already has? Thanks.

**Comment by simolus3** [comment]
@jims57 I can't reproduce this issue with the latest version. Which version of `moor_generator` are you using?

**Comment by jims57** [comment]
> @jims57 I can't reproduce this issue with the latest version. Which version of `moor_generator` are you using?

Thank for your reply, I am using `moor_generator: ^3.4.0` in our project.

By the way, please reply the previous reply:

> > change the migration logic in your database to not create this table of course
> 
> By the way, how to make **Moor** not to create **sqlite_sequence** table in migration logic?

**Comment by simolus3** [comment]
Ok, so the approach of using a compiled custom query should work with the latest moor version (4.4). With that you wouldn't have to declare a `SqliteSequence` table either.

> By the way, how to make Moor not to create sqlite_sequence table in migration logic?

Simply avoid using `Migrator.createAll()`. Instead, use

```dart
onCreate: (m) async {
  for (final entity in allSchemaEntities) {
     if (entity != sqliteSequence) await m.create(entity);
  }
}
```

**Comment by jims57** [comment]
@simolus3 Thank for your support.

I decided to use your first approach to make it. Here is my code:

![image](https://user-images.githubusercontent.com/1920873/133044563-a1fe2e95-950f-4f8b-a4ed-2eabec828e0e.png)

It works. But issueCustomQuery is shown as deprecated, will it be caused potential risks in the future?

By the way, I don't plan to upgrade Moor to the latest version, because after I upgrade it, it requires newer version of Dart in my project, and then I have to upgrade Flutter or something. Last time, we upgraded Dart and Flutter, it caused our project fail to build, spending us a lots of time to figure out why it happened(failed to build, since our project is urgent to deploy, we planned to find ways in the next release of our project to fix the problem, but no this version.).

In our project, it is using `moor: ^3.4.0` , so **await m.create(entity);** isn't supportive, is it the method in Moor 4.5?

**Comment by simolus3** [comment]
> But issueCustomQuery is shown as deprecated, will it be caused potential risks in the future?

You can just call `customStatement` on your database in your moor version, you don't have to call it on the migrator.

And I see your point about upgrading Flutter, it can be a bit daunting. The latest moor version is using some language features only available in Dart 2.13, so we had to add the min constraint.

> so await m.create(entity); isn't supportive, is it the method in Moor 4.5?

Yes, but your if/else if check over all entities should work as well :+1:
